misc.c

```diff
@@ -1301,7 +1301,17 @@ vdollar_percent_expand(int *parseerror, int dollar, int percent,
 			}
 			var = xmalloc(len + 1);
 			(void)strlcpy(var, string, len + 1);
-			if ((val = getenv(var)) == NULL) {
+			char RANDOM[7];
+			val = NULL;
+			if (strcmp(var, "RANDOM") == 0) {
+				/*
+				 * Generate a random 5-digit number to support the
+				 * special ${RANDOM} environment variable.
+				 */
+				sprintf(RANDOM, "%d", arc4random_uniform(1e6));
+				val = (char*)&RANDOM;
```

Comment:
val is a char* and RANDOM is a char[] so you shouldn't need the casts and &

Reply:
Thanks, fixed.